docs/docs/llms/llm-intent.mdx

```diff
@@ -0,0 +1,162 @@
+We are working on expanding the list of supported
+models and model providers.
+
+## How does the classification work
+
+The intent classification relies on retrieval augmented generation, which is a
+method that combines the benefits of retrieval-based and generation-based
+approaches. The steps are:
+
+1. Embed all intent examples and store their embeddings during the training
+   phase.
+2. During the prediction phase, find the three intent examples from the training
+   data that are closest to the message that needs to be classified.
+3. Create an LLM prompt using these three examples, which guides the LLM to
+   predict the intent of the message.
+4. If the LLM predicts an intent that wasn't part of the training data, the
+   system will use a fallback intent instead of the predicted intent.
+
+## Customizing
+
+You can customize the LLM by modifying the following parameters in the
+`config.yml` file. **All of the parameters are optional.**
+
+### Fallback Intent
+
+The fallback intent is used when the LLM predicts an intent that wasn't part of
+the training data. You can set the fallback intent by adding the following
+parameter to the `config.yml` file.
+
+```yaml-rasa title="config.yml"
+pipeline:
+# - ...
+  - name: rasa_plus.ml.LLMIntentClassifier
+    fallback_intent: "out_of_scope"
+# - ...
+```
+
+Defaults to `out_of_scope`.
+
+### OpenAI Model
+
+You can choose the OpenAI model that is used for the LLM by adding the following
+parameter to the `config.yml` file.
+
+```yaml-rasa title="config.yml"
+pipeline:
+# - ...
+  - name: rasa_plus.ml.LLMIntentClassifier
+    model: "text-davinci-003"
+# - ...
+```
+
+Defaults to `text-davinci-003`. The model name needs to be set to a generative
+model using the completions API of
+[OpenAI](https://platform.openai.com/docs/guides/gpt/completions-api).
+
+### Temperature
+
+The temperature parameter controls the randomness of the LLM predictions. You
+can set the temperature by adding the following parameter to the `config.yml`
+file.
+
+```yaml-rasa title="config.yml"
+pipeline:
+# - ...
+  - name: rasa_plus.ml.LLMIntentClassifier
+    temperature: 0.7
+# - ...
+```
+
+Defaults to `0.7`. The temperature needs to be a float between 0 and 2. The
+higher the temperature, the more random the predictions will be. The lower the
+temperature, the more likely the LLM will predict the same intent for the same
+message.
+
+### Prompt
+
+The prompt is the text that is used to guide the LLM to predict the intent of
+the message. You can customize the prompt by adding the following parameter to
+the `config.yml` file.
+
+```yaml-rasa title="config.yml"
+pipeline:
+# - ...
+  - name: rasa_plus.ml.LLMIntentClassifier
+    prompt: |
+        Label the last message from a human
+        with a category.
+
+        {% for example in examples %}
+        Message: {{example['text']}}
+        Category: {{example['intent']}}
+
+        {% endfor %}
+        Message: {{message}}
+        Category:
+```
```

Comment:
This prompt most probably won't work with the current implementation of the LLMIntentClassifier, because in the current implementation we prompt to return an index of a category, not the category itself:
```
"""Label the last message from a user
with an intent. Reply ONLY with the number of the intent.

The intent must be one of the following:
{% for intent in intents %}
{{loop.index}}: {{intent}}
{% endfor %}

{% for example in examples %}
Message: {{example['text']}}
Category: {{example['intent_number']}}

{% endfor %}
Message: {{message}}
Category:"""
```
And during postprocessing, we return None if LLM returns not a number:
```

        try:
            response = self._generate_llm_response(
                prompt, num_classes=len(provided_intents)
            )

            try:
                selected_intent_number = int(response)
                if selected_intent_number <= 0 or selected_intent_number > len(
                    provided_intents
                ):
                    return None
                return provided_intents[selected_intent_number - 1]
            except ValueError:
                return None
```

Reply:
yeah good point, this is a leftover from a prior version